### PR TITLE
feature: Add E2E test base file for sealer

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it
Adding E2e test for sealer apply to keep our code more robust

### Describe how you did it
1, split github work flow as more jobs
2, add test framework for sealer ,will run sealer as non-root user.
3, add utils file and fixtures 

### Describe how to verify it

install ginkgo and config your env as readme.md said.
ginkgo -v --focus="sealer apply" test

### CODE REVIEW

File: test/suites/build/build.go [hunk 1]
```diff
@@ -1,22 +1,99 @@
...
+
+func GetTestClusterFilePath() string {
+	return ""
+}
+
+type ArgsOfBuild struct {
+	KubeFile, ImageName, Context, BuildType string
+}
+
+func (a *ArgsOfBuild) SetKubeFile(kubeFile string) *ArgsOfBuild {
```
Comment (justadogistaken) [inline test/suites/build/build.go:51]: attribution could be set using "a.KubeFile = kubeFile", why encapsulate the operation?

File: test/suites/build/build.go [hunk 2]
```diff
@@ -1,22 +1,99 @@
...
+	a.Context = context
+	return a
+}
+
+func (a *ArgsOfBuild) SetBuildType(buildType string) *ArgsOfBuild {
+	a.BuildType = buildType
+	return a
+}
+
+func (a *ArgsOfBuild) ToString() string {
```
Comment (justadogistaken) [inline test/suites/build/build.go:71]: If you do the operations above for implementing ArgsOfBuild in Builder mode, "ToString()" to "Build()" would be better, or just String(). just a suggestion.

File: test/suites/build/build.go [hunk 3]
```diff
@@ -1,22 +1,99 @@
...
+	return fmt.Sprintf("%s build -f %s -t %s -c %s -b %s", settings.DefaultSealerBin, a.KubeFile, a.ImageName, a.Context, a.BuildType)
```
Comment (justadogistaken) [inline test/suites/build/build.go:72]: Can we guarantee those params are not empty? if not so, maybe validate them first would be nicer.

File: test/suites/build/build.go [hunk 4]
```diff
@@ -1,22 +1,99 @@
...
+}
+
+func NewArgsOfBuild() *ArgsOfBuild {
+	return &ArgsOfBuild{}
```
Comment (justadogistaken) [inline test/suites/build/build.go:76]: NewArgsOfBuild is too simple, I think let call do "&ArgsOfBuild{}" would be fine.

File: test/testhelper/utils.go [hunk 5]
```diff
@@ -1,13 +1,82 @@
...
 }
+
+func CreateTempFile() string {
+	dir := os.TempDir()
+	file, err := ioutil.TempFile(dir, "tmpfile")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	defer func() {
+		err := file.Close()
+		if err != nil {
+			fmt.Println(err)
```
Comment (justadogistaken) [inline test/testhelper/utils.go:30]: Is there any reason for fmt.Println(err) instead of logger?

File: test/suites/build/build.go (hunk 2 above)
Comment (kakaZhou719) [reply test/suites/build/build.go:85]: truely, seem like  "Build()"  more better than "ToString()"

File: test/suites/build/build.go (hunk 3 above)
Comment (kakaZhou719) [reply test/suites/build/build.go:86]: yes, need to do validate .

File: test/suites/build/build.go (hunk 1 above)
Comment (fanux) [reply test/suites/build/build.go:49]: I think that's okay, because you don't know the set logic will change or not in the future.

**Review by justadogistaken — APPROVED** [review]
LGTM if you've tested already

**Pull request merged**